include/circt/Dialect/SV/SVStatements.td

```diff
@@ -138,6 +138,60 @@ def AlwaysOp : SVOp<"always", [HasRegionTerminator, NoRegionArguments,
   let verifier = "return ::verifyAlwaysOp(*this);";
 }
 
+def NoReset: I32EnumAttrCase<"NoReset", 0, "noreset">;
+def SyncReset: I32EnumAttrCase<"SyncReset", 1, "syncreset">;
+def AsyncReset: I32EnumAttrCase<"AsyncReset", 2, "asyncreset">;
```

Comment:
What about the case where you've got an asynchronous reset assert, and a synchronous de-assert? This should be relatively common as I think it's generally recommended for performant reset trees.

Reply:
For synchronized asynchronous resets, these can be modeled as two back-to-back registers (one async, one sync), or by adding a new enum type.  But this is simple to add at a future point when needed.

Reply:
The back-to-back registers should work fine here. I don't think that this has to be included in the type of the `always_ff` reset because that should have no effect on the actual emitted always block. I.e., there's three emission styles here:

- "No reset"
```verilog
always_ff @(<edge> <clock>) begin
  /* ... */
end 
```

- "Synchronous Reset"
```verilog
always_ff @(<edge> <clock>) begin
  if (<reset>) begin
    /* ... */
  end
  else begin
    /* ... */
  end
end
```

- "Asynchronous Reset"
```verilog
always_ff @(<edge> <clock> or <edge> <reset>) begin
  if (<reset>) begin
    /* ... */
  end
  else begin
    /* ... */
  end
end
```

The back-to-back registers for de-asserting reset would assumedly be in some other module and wouldn't affect emission here. (We probably don't want such reset synchronization logic on every always block?) This does bring up a great question if and how reset metadata _that doesn't affect local Verilog emission_ should be tracked.

Reply:
I see the value in folding the async-sync case in as we then track semantics, but since it's not required and is easy to add, I'm all for pushing this off until we have clarity on register representation (where it rightly belongs).  I don't want to always_ff to be our register representation.